File: src/cs/Schematron.Insight/Phase.cs
using System;
using System.Collections.Generic;

namespace Schematron.Insight
{
    public class Phase
    {
        public static string ALL = "#ALL";
        public static string DEFAULT = "#DEFAULT";
        public string Id { get; set; } = "";
        public List<string> Patterns { get; set; } = new List<string>();
        public bool IsEnabled => !String.IsNullOrWhiteSpace(Id) && Patterns.Count > 0;

    }
}

File: src/java/Schematron.Insight/src/main/java/com/ehirakawa/schkit/Phase.java
package com.ehirakawa.schkit;

import java.util.ArrayList;

import org.apache.commons.lang3.StringUtils;

public class Phase {
	public static final String ALL = "#ALL";
	public static final String DEFAULT = "#DEFAULT";

	private String id = "";
	private ArrayList<String> patterns = new ArrayList<String>();

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public ArrayList<String> getPatterns() {
		return patterns;
	}

	public void setPatterns(ArrayList<String> patterns) {
		this.patterns = patterns;
	}

	public boolean isEnabled() {
		return !StringUtils.isBlank(id) && patterns.size() > 0;
	}
}
